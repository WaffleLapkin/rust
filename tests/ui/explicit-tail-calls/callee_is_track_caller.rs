#![expect(incomplete_features)]
#![feature(explicit_tail_calls)]

fn a() {
    become b(); //~ error: a function marked with `#[track_caller]` cannot be tail-called
}

#[track_caller]
fn b() {}

fn main() {}
