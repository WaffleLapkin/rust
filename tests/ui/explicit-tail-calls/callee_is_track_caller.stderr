error: a function marked with `#[track_caller]` cannot be tail-called
  --> $DIR/callee_is_track_caller.rs:5:5
   |
LL |     become b();
   |     ^^^^^^^^^^

error: aborting due to 1 previous error

